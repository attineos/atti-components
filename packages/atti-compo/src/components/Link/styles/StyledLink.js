import styled from 'styled-components'

const StyledLink = styled.a`
  color: ${({ theme }) => theme.components.link.colors.text};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

export default StyledLink
