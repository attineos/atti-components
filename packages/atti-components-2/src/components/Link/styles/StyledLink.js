import styled, { css } from 'styled-components'

const colors = css`
  color: ${({ theme }) => theme.components.link.colors.text};
`

// TODO: should inherit its font values when other components (like Paragraph) will be ready
const fonts = css`
  font-size: ${({ theme }) => theme.components.link.fonts.fontSize};
  font-weight: ${({ theme }) => theme.components.link.fonts.fontWeight};
  font-family: ${({ theme }) => theme.components.link.fonts.fontFamily};
`

const StyledLink = styled.a`
  ${colors};
  ${fonts};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

export default StyledLink
