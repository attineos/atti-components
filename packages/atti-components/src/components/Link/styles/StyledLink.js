import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import { Text } from '../../Typographies'

const colors = css`
  color: ${({ theme }) => theme.components.link.colors.text};
`

const StyledLink = styled(Text.withComponent('a'))`
  ${colors};
  text-decoration: none;
  border-bottom: ${({ theme }) => theme.components.link.borders.width} solid
    ${({ theme }) => theme.components.link.colors.border};

  &:hover {
    border-color: ${({ theme }) => theme.components.link.colors.borderHover};
  }
`

StyledLink.propTypes = {
  /**
   * Text of the link.
   */
  children: PropTypes.string.isRequired,
}

export default StyledLink
