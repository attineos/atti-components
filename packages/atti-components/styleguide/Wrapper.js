import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
