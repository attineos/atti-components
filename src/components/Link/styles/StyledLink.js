import styled from 'styled-components'
import Text from '../../Text'

const StyledLink = styled(Text.withComponent('a'))`
  color: ${({ theme }) => theme.components.link.colors.text};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

export default StyledLink
