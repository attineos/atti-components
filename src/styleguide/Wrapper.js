import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
