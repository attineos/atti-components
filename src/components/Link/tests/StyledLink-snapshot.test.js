/* This file was generated automatically, please do not write in it. */
import React from 'react';
import StyledLink from '../styles/StyledLink';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('StyledLink renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <StyledLink ></StyledLink>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    