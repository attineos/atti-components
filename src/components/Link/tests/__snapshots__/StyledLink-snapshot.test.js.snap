// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink renders correctly 1`] = `
<span
  className="atti-box"
>
  <span
    className="atti-box"
  >
    <a
      className="Text-sc-187vq02-0-a StyledLink-kwdrvq-0 atti-r-css  iKaJKm"
    >
      Atticomponents
    </a>
  </span>
</span>
`;
